### PR TITLE
Move swiftpm to swift_build_support infra

### PR DESCRIPTION
**Pull request by aciidgh**
This will allow cleaning up most of the hacks in SwiftPM's build script.

<rdar://problem/56220087>

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

File: utils/swift_build_support/swift_build_support/products/swiftpm.py [hunk 1]
```diff
@@ -2,16 +2,81 @@
...
+        if action != "build":
+            helper_cmd.append(action)
+
+        if self.is_release():
+            helper_cmd.append("--release")
+
+        helper_cmd += [
+            "--swiftc=" + swiftc,
+            "--sbt=" + sbt,
+            "--build", self.build_dir,
```
Comment (ahoppen) [inline utils/swift_build_support/swift_build_support/products/swiftpm.py]: Any reason why all the other commands are `=` separated and this one is space separated?

File: utils/swift_build_support/swift_build_support/products/swiftpm.py [hunk 2]
```diff
@@ -2,16 +2,81 @@
...
+            "--llbuild-source-dir=" + llbuild_src,
+            "--llbuild-build-dir=" + llbuild_build_dir,
+        ]
+        helper_cmd.extend(additional_params)
+
+        shell.call(helper_cmd)
```
Comment (ahoppen) [inline utils/swift_build_support/swift_build_support/products/swiftpm.py]: If this command takes some time, you can also call `call_without_sleeping` which caffeinates the machine during the invocation. I meant to change this for all the other build products as well.

**Review by ahoppen — COMMENTED** [review]
Looking good overall, just some minor comments inline. 

As discussed offline, I think it would be nice to remove the old logic from `build-script-impl` to not have dead code lying around and see the difference in code size for build script logic.

File: utils/swift_build_support/swift_build_support/products/swiftpm.py [hunk 3]
```diff
@@ -2,16 +2,81 @@
...
+            self.source_dir, 'Utilities', 'bootstrap')
+        toolchain_path = self.install_toolchain_path()
+        swiftc = os.path.join(toolchain_path, "usr", "bin", "swiftc")
+        sbt = os.path.join(toolchain_path, "usr", "bin", "swift-build-tool")
+
+        llbuild_src = os.path.join(
+             os.path.dirname(self.source_dir), "llbuild")
+
+        # FIXME: We require llbuild build directory in order to build. Is
+        # there a better way to get this?
```
[the remark this answers is not in the document]
Comment (aciidb0mb3r) [reply utils/swift_build_support/swift_build_support/products/swiftpm.py]: llbuild is indeed installed but without the headers. We don't want to ship llbuild's header in the toolchain so we definitely need access to its build directory.

File: utils/swift_build_support/swift_build_support/products/swiftpm.py (hunk 1 above)
Comment (aciidb0mb3r) [reply utils/swift_build_support/swift_build_support/products/swiftpm.py]: I don't think so. I'll check.

File: utils/swift_build_support/swift_build_support/products/swiftpm.py (hunk 2 above)
Comment (aciidb0mb3r) [reply utils/swift_build_support/swift_build_support/products/swiftpm.py]: Is that correct though? Maybe build script should re-exec itself with caffeinate

File: utils/swift_build_support/swift_build_support/products/swiftpm.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ahoppen) [reply utils/swift_build_support/swift_build_support/products/swiftpm.py]: Stupid question but couldn’t you find the headers in the source directory? I just looked at a local builds of mine and there weren't any headers in llbuild’s build directory.

File: utils/swift_build_support/swift_build_support/products/swiftpm.py (hunk 2 above)
Comment (ahoppen) [reply utils/swift_build_support/swift_build_support/products/swiftpm.py]: True. You can just leave it as `shell.call` for now if you want.

File: utils/swift_build_support/swift_build_support/products/swiftpm.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (aciidb0mb3r) [reply utils/swift_build_support/swift_build_support/products/swiftpm.py]: ah sorry, I was referring to Swift "headers" aka the `.swiftmodule` file.

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

File: utils/swift_build_support/swift_build_support/products/swiftpm.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ahoppen) [reply utils/swift_build_support/swift_build_support/products/swiftpm.py]: Ah, I see. It’s good with me then.

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test linux

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Review by ahoppen — APPROVED** [review]
Looking good now!

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test linux

**Comment by aciidb0mb3r** [comment]
@swift-ci build toolchain linux

**Comment by aciidb0mb3r** [comment]
macOS toolchain builder succeeded but the status check wasn't updated. See: https://ci.swift.org/job/swift-PR-toolchain-osx/426/

**Comment by aciidb0mb3r** [comment]
Tested both mac and linux toolchains.

**Pull request merged**